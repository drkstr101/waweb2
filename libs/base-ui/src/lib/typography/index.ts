export { default as typography } from './styles';
export type { TypographyCSS } from './styles/TypographyCSS';
export * from './typography';
export type {
  BodyProps,
  CodeProps,
  DetailProps,
  HeadingProps,
  TypographyComponent,
  TypographyStyle,
  TypographyVariant
} from './typography';
