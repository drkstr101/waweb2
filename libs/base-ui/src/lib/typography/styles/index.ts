import styles from './typography.module.css';
import { TypographyCSS } from './TypographyCSS';

const _styles: TypographyCSS = styles as TypographyCSS;
export default _styles;

// .spectrum-Article
// .spectrum-Article .spectrum-Body--large
// .spectrum-Article .spectrum-Body--secondary
// .spectrum-Article .spectrum-Body--small
// .spectrum-Article .spectrum-Body1
// .spectrum-Article .spectrum-Body2
// .spectrum-Article .spectrum-Body3
// .spectrum-Article .spectrum-Body4
// .spectrum-Article .spectrum-Body5
// .spectrum-Article .spectrum-Detail
// .spectrum-Article .spectrum-Heading--display
// .spectrum-Article .spectrum-Heading--pageTitle
// .spectrum-Article .spectrum-Heading--subtitle1
// .spectrum-Article .spectrum-Heading--subtitle2
// .spectrum-Article .spectrum-Heading--subtitle3
// .spectrum-Article .spectrum-Heading1
// .spectrum-Article .spectrum-Heading1--display
// .spectrum-Article .spectrum-Heading1--display.spectrum-Heading1--quiet
// .spectrum-Article .spectrum-Heading1--quiet
// .spectrum-Article .spectrum-Heading2
// .spectrum-Article .spectrum-Heading2--display
// .spectrum-Article .spectrum-Heading2--display.spectrum-Heading2--quiet
// .spectrum-Article .spectrum-Heading2--quiet
// .spectrum-Article .spectrum-Heading3
// .spectrum-Article .spectrum-Heading4
// .spectrum-Article .spectrum-Heading5
// .spectrum-Article .spectrum-Heading6
// .spectrum-Article .spectrum-Subheading
// .spectrum-Body
// .spectrum-Body--italic
// .spectrum-Body--large
// .spectrum-Body--secondary
// .spectrum-Body--small
// .spectrum-Body1
// .spectrum-Body2
// .spectrum-Body3
// .spectrum-Body4
// .spectrum-Body5
// .spectrum-Code1
// .spectrum-Code2
// .spectrum-Code3
// .spectrum-Code4
// .spectrum-Code5
// .spectrum-Detail
// .spectrum-Heading--display
// .spectrum-Heading--pageTitle
// .spectrum-Heading--subtitle1
// .spectrum-Heading--subtitle2
// .spectrum-Heading--subtitle3
// .spectrum-Heading1
// .spectrum-Heading1--display
// .spectrum-Heading1--display.spectrum-Heading1--quiet
// .spectrum-Heading1--display.spectrum-Heading1--strong
// .spectrum-Heading1--quiet
// .spectrum-Heading1--strong
// .spectrum-Heading2
// .spectrum-Heading2--display
// .spectrum-Heading2--display.spectrum-Heading2--quiet
// .spectrum-Heading2--display.spectrum-Heading2--strong
// .spectrum-Heading2--quiet
// .spectrum-Heading2--strong
// .spectrum-Heading3
// .spectrum-Heading4
// .spectrum-Heading5
// .spectrum-Heading6
// .spectrum-Subheading
// .spectrum-Typography
// .spectrum-Typography .spectrum-Body--large
// .spectrum-Typography .spectrum-Body--secondary
// .spectrum-Typography .spectrum-Body--small
// .spectrum-Typography .spectrum-Body1
// .spectrum-Typography .spectrum-Body2
// .spectrum-Typography .spectrum-Body3
// .spectrum-Typography .spectrum-Body4
// .spectrum-Typography .spectrum-Body5
// .spectrum-Typography .spectrum-Detail
// .spectrum-Typography .spectrum-Heading--display
// .spectrum-Typography .spectrum-Heading--pageTitle
// .spectrum-Typography .spectrum-Heading--subtitle1
// .spectrum-Typography .spectrum-Heading--subtitle2
// .spectrum-Typography .spectrum-Heading--subtitle3
// .spectrum-Typography .spectrum-Heading1
// .spectrum-Typography .spectrum-Heading1--display
// .spectrum-Typography .spectrum-Heading1--display.spectrum-Heading1--quiet
// .spectrum-Typography .spectrum-Heading1--display.spectrum-Heading1--strong
// .spectrum-Typography .spectrum-Heading1--quiet
// .spectrum-Typography .spectrum-Heading1--strong
// .spectrum-Typography .spectrum-Heading2
// .spectrum-Typography .spectrum-Heading2--display
// .spectrum-Typography .spectrum-Heading2--display.spectrum-Heading2--quiet
// .spectrum-Typography .spectrum-Heading2--display.spectrum-Heading2--strong
// .spectrum-Typography .spectrum-Heading2--quiet
// .spectrum-Typography .spectrum-Heading2--strong
// .spectrum-Typography .spectrum-Heading3
// .spectrum-Typography .spectrum-Heading4
// .spectrum-Typography .spectrum-Heading5
// .spectrum-Typography .spectrum-Heading6
// .spectrum-Typography .spectrum-Subheading
