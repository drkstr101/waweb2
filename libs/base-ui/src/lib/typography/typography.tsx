import clsx from 'clsx';
import { HtmlHTMLAttributes } from 'react';
import styles from './styles';

export type TypographyVariant = 'body' | 'heading' | 'detail' | 'code';
export type TypographyStyle = 'light' | 'strong' | 'heavy' | 'emphasized';
export type TypographyComponent =
  | 'div'
  | 'p'
  | 'span'
  | 'pre'
  | 'code'
  | 'h1'
  | 'h2'
  | 'h3'
  | 'h4'
  | 'h5'
  | 'h6';

export interface TypographyProps extends HtmlHTMLAttributes<HTMLDivElement> {
  component?: TypographyComponent;
  font?: TypographyStyle;
  serif?: boolean;
}

function Typography({ className, children, ...props }: TypographyProps) {
  console.log(styles);
  const Component = props.component ?? 'p';
  return <Component className={className}>{children}</Component>;
}

export interface HeadingProps extends Omit<TypographyProps, 'className'> {
  size?: 'XXS' | 'XS' | 'S' | 'M' | 'L' | 'XL' | 'XXL' | 'XXXL';
  variant?: 'heading' | 'display' | 'title' | 'subtitle1' | 'subtitle2' | 'subtitle3';
}

export function Heading(props: HeadingProps) {
  const { size = 'M', font = 'heavy', serif = true } = props;
  const className = clsx({
    [styles['spectrum-Heading']]: true,
    [styles['spectrum-Heading--emphasized']]: font === 'emphasized',
    [styles['spectrum-Heading--heavy']]: font === 'heavy',
    [styles['spectrum-Heading--strong']]: font === 'strong',
    [styles['spectrum-Heading--light']]: font === 'light',
    [styles['spectrum-Heading--serif']]: serif,
    [styles['spectrum-Heading--sizeXXS']]: size === 'XXS',
    [styles['spectrum-Heading--sizeXS']]: size === 'XS',
    [styles['spectrum-Heading--sizeS']]: size === 'S',
    [styles['spectrum-Heading--sizeM']]: size === 'M',
    [styles['spectrum-Heading--sizeL']]: size === 'L',
    [styles['spectrum-Heading--sizeXL']]: size === 'XL',
    [styles['spectrum-Heading--sizeXXL']]: size === 'XXL',
    [styles['spectrum-Heading--sizeXXXL']]: size === 'XXXL',
    [styles['spectrum-Typography']]: true
  });

  return <Typography className={className} {...props} />;
}

export interface BodyProps extends Omit<TypographyProps, 'className'> {
  size?: 'XS' | 'S' | 'M' | 'L' | 'XL' | 'XXL' | 'XXXL';
}

export function Body(props: BodyProps) {
  const { size = 'M', serif = false } = props;
  const className = clsx({
    [styles['spectrum-Typography']]: true,
    [styles['spectrum-Body']]: true,
    [styles['spectrum-Body--serif']]: serif,
    [styles['spectrum-Body--sizeL']]: size === 'L',
    [styles['spectrum-Body--sizeM']]: size === 'M',
    [styles['spectrum-Body--sizeXS']]: size === 'XS',
    [styles['spectrum-Body--sizeS']]: size === 'S',
    [styles['spectrum-Body--sizeXL']]: size === 'XL',
    [styles['spectrum-Body--sizeXXL']]: size === 'XXL',
    [styles['spectrum-Body--sizeXXXL']]: size === 'XXXL'
  });

  return <Typography className={className} {...props} />;
}

export interface DetailProps extends Omit<TypographyProps, 'className'> {
  size?: 'S' | 'M' | 'L' | 'XL';
}

export function Detail(props: DetailProps) {
  const { size = 'N', font, serif = false } = props;
  const className = clsx({
    [styles['spectrum-Typography']]: true,
    [styles['spectrum-Detail']]: true,
    [styles['spectrum-Detail--emphasized']]: font === 'emphasized',
    [styles['spectrum-Detail--light']]: font === 'light',
    [styles['spectrum-Detail--strong']]: font === 'strong',
    [styles['spectrum-Detail--serif']]: serif,
    [styles['spectrum-Detail--sizeS']]: size === 'S',
    [styles['spectrum-Detail--sizeM']]: size === 'M',
    [styles['spectrum-Detail--sizeL']]: size === 'L',
    [styles['spectrum-Detail--sizeXL']]: size === 'XL'
  });

  return <Typography className={className} {...props} />;
}

export interface CodeProps extends Omit<TypographyProps, 'className'> {
  size?: 'XS' | 'S' | 'M' | 'L' | 'XL';
}

export function Code(props: CodeProps) {
  const { size = 'N', serif = false } = props;
  const className = clsx({
    [styles['spectrum-Typography']]: true,
    [styles['spectrum-Code']]: true,
    [styles['spectrum-Code--serif']]: serif,
    [styles['spectrum-Code--sizeXS']]: size === 'XS',
    [styles['spectrum-Code--sizeS']]: size === 'S',
    [styles['spectrum-Code--sizeL']]: size === 'L',
    [styles['spectrum-Code--sizeM']]: size === 'M',
    [styles['spectrum-Code--sizeXL']]: size === 'XL'
  });

  return <Typography className={className} {...props} />;
}
