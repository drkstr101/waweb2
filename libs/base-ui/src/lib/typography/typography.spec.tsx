import { render } from '@testing-library/react';
import { Body, Code, Detail, Heading } from './typography';

describe('watheia.waweb/base-ui.typography', () => {
  it('Heading should render successfully', () => {
    const { container } = render(<Heading component="h1">Hello, World!</Heading>);
    expect(container).toBeInstanceOf(HTMLElement);
    expect(container).toMatchSnapshot();
  });
  it('Body should render successfully', () => {
    const { container } = render(<Body>Hello, World!</Body>);
    expect(container).toBeInstanceOf(HTMLElement);
    expect(container).toMatchSnapshot();
  });
  it('Heading should render successfully', () => {
    const { container } = render(<Code component="pre">Hello, World!</Code>);
    expect(container).toBeInstanceOf(HTMLElement);
    expect(container).toMatchSnapshot();
  });
  it('Heading should render successfully', () => {
    const { container } = render(<Detail>Hello, World!</Detail>);
    expect(container).toBeInstanceOf(HTMLElement);
    expect(container).toMatchSnapshot();
  });
});
