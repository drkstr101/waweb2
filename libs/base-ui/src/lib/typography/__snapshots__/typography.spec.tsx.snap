// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`watheia.waweb/base-ui.typography Body should render successfully 1`] = `
<div>
  <p
    class="spectrum-Typography spectrum-Body spectrum-Body--sizeM"
  >
    Hello, World!
  </p>
</div>
`;

exports[`watheia.waweb/base-ui.typography Heading should render successfully 1`] = `
<div>
  <h1
    class="spectrum-Heading spectrum-Heading--heavy spectrum-Heading--serif spectrum-Heading--sizeM spectrum-Typography"
  >
    Hello, World!
  </h1>
</div>
`;

exports[`watheia.waweb/base-ui.typography Heading should render successfully 2`] = `
<div>
  <pre
    class="spectrum-Typography spectrum-Code"
  >
    Hello, World!
  </pre>
</div>
`;

exports[`watheia.waweb/base-ui.typography Heading should render successfully 3`] = `
<div>
  <p
    class="spectrum-Typography spectrum-Detail"
  >
    Hello, World!
  </p>
</div>
`;
